Add range sliders and range selection buttons for date zooming

jeremymoreau opened the pull request:
Adds date range selection sliders and buttons.
<img width="625" alt="range sliders" src="https://user-images.githubusercontent.com/7446564/104836201-79034580-5869-11eb-8a32-932ad0f4d853.png">

jeremymoreau committed:
Add range sliders and range selection buttons for date zooming

jeremymoreau committed:
Fix ytd FR label

mschoettle commented:
This is awesome! This allows to focus only on the second wave for example.

Could be interesting to add one additional smaller time frame in there. What do you think about 7 or 14 days?

And, I suggest to either rename the function `add_ylog_menu` to something more appropriate (since it now does more than that one thing). Or maybe split them into separate functions. I find it difficult for the figures with different lines to look at the log because it will only change one of them. That's why I had disabled it for the testing figure (plus it uses the negative tests).

jeremymoreau committed:
Merge branch 'master' into add-range-sliders

jeremymoreau committed:
Add 14days range and rename `add_ylog_menu()`

jeremymoreau committed:
Disable reset on tap/double-click

On mobile, switching to a range (e.g. 14 days) and then tapping to select a date and display the hover labels would instead reset the plot to the default range.
